### PR TITLE
Improve configure xcrun for iOS

### PR DESCRIPTION
**Pull request by cntrump**
Remove flag `-mios-version-min=7.0.0`, using `export IPHONEOS_DEPLOYMENT_TARGET=7.0`.
Disable "dso", "hw", "engine", "tests"

clang can be overwrited by `export CC`

Example: build  iPhone simulator for Apple Silicon

```bash
export CROSS_TARGET=iPhoneSimulator
export CROSS_SDK=${CROSS_TARGET}.sdk
export CROSS_TOP=`xcode-select -p`/Platforms/${CROSS_TARGET}.platform/Developer
export CC=`xcrun --find clang`
export CFLAGS="-flto -mios-simulator-version-min=9.0 -arch arm64"
export LDFLAGS="-flto -mios-simulator-version-min=9.0 -arch arm64"

./Configure iphoneos-cross
make -j
```

Example 2: build with xcrun

```bash
export IPHONEOS_DEPLOYMENT_TARGET=9.0
```

iPhone Simulator

```bash
export CFLAGS="-arch arm64"  # Intel chip and Apple Silicon
./Configure iossimulator-xcrun
make -j
```

iPhone OS arm64 (armv7 is deprecated)

```bash
./Configure ios64-xcrun
make -j
```

### CODE REVIEW

**Comment by cntrump** [comment]
CLA: trivial

File: Configurations/15-ios.conf [hunk 1]
```diff
@@ -17,22 +17,22 @@ my %targets = (
         # It should be possible to go below iOS 6 and even add -arch armv6,
         # thus targeting iPhone pre-3GS, but it's assumed to be irrelevant
         # at this point.
-        CC               => "xcrun -sdk iphoneos cc",
```
Comment (t8m) [inline Configurations/15-ios.conf:20]: it is unclear if dropping things as xcrun -sdk iphoneos is a good idea.

Comment (cntrump) [reply Configurations/15-ios.conf:20]: > it is unclear if dropping things as xcrun -sdk iphoneos is a good idea.

Commit updated, `xcrun -sdk iphoneos cc` is a wrong command.

**Comment by t8m** [comment]
It would be really good to have some independent review who builds OpenSSL for iOS here.

**Comment by cntrump** [comment]
Now, build `OpenSSL.xcframework` for iOS is so easy.

build-ios-xcframework.sh

```bash
#!/usr/bin/env zsh

set -e

target=$(cd "$(dirname "$0")";pwd)

[ -d "${target}/out" ] && rm -rf "${target}/out"

build_openssl_lib() {
    ./Configure $@

    make clean
    make -j
    make install
}

write_openssl_modulemap() {
    echo "framework module OpenSSL [extern_c] {" > $1
    echo "    umbrella header \"OpenSSL.h\"" >> $1
    echo "    export *" >> $1
    echo "    module * { export * }" >> $1
    echo "}" >> $1
}

build_openssl_framework() {
    pushd "$1"
    rm -rf bin share ssl
    mv include/openssl Headers
    rm -rf include
    libtool -static -o OpenSSL lib/libcrypto.a lib/libssl.a
    rm -rf lib
    mkdir Modules
    write_openssl_modulemap Modules/module.modulemap

    pushd Headers
    for h in *.h
    do
        if [ "${h}" = "asn1_mac.h" ]; then
            continue
        fi

        echo "#import <OpenSSL/${h}>" >> ../OpenSSL.h
    done
    mv ../OpenSSL.h ./
    popd

    popd
}

cleanup() {
    rm -rf $@
}

xcrun_clang=$(xcrun --find clang)

build_openssl_lib --prefix="${target}/out/ios-arm64/OpenSSL.framework" \
                    ios64-xcrun \
                    IPHONEOS_DEPLOYMENT_TARGET=9.0 \
                    CC=${xcrun_clang}

build_openssl_lib --prefix="${target}/out/ios-arm64_x86_64-simulator/OpenSSL.framework" \
                    iossimulator-xcrun \
                    IPHONEOS_DEPLOYMENT_TARGET=9.0 \
                    CC=${xcrun_clang} \
                    CFLAGS="-arch arm64"

build_openssl_framework "${target}/out/ios-arm64/OpenSSL.framework"
build_openssl_framework "${target}/out/ios-arm64_x86_64-simulator/OpenSSL.framework"

xcodebuild -create-xcframework \
           -framework "${target}/out/ios-arm64/OpenSSL.framework" \
           -framework "${target}/out/ios-arm64_x86_64-simulator/OpenSSL.framework" \
           -output \
           "${target}/out/OpenSSL.xcframework"

cleanup "${target}/out/ios-arm64" \
        "${target}/out/ios-arm64_x86_64-simulator"
```

File: Configurations/15-ios.conf [hunk 2]
```diff
@@ -53,12 +70,10 @@ my %targets = (
...
-        CC               => "cc",
         cflags           => add("-isysroot \$(CROSS_TOP)/SDKs/\$(CROSS_SDK)"),
     },
     "ios64-cross" => {
         inherit_from     => [ "ios64-xcrun" ],
-        CC               => "cc",
         cflags           => add("-isysroot \$(CROSS_TOP)/SDKs/\$(CROSS_SDK)"),
     },
-);
+);
```
Comment (paulidale) [inline Configurations/15-ios.conf:79]: Please include a new line at the end of the file.

**Comment by paulidale** [comment]
It's not clear if this is a bug fix or a feature.  A feature is only eligible for master, a bug fix could be back ported to 3.0 in addition.

File: Configurations/15-ios.conf (hunk 2 above)
Comment (cntrump) [reply Configurations/15-ios.conf]: done.

**Comment by cntrump** [comment]
> It's not clear if this is a bug fix or a feature. A feature is only eligible for master, a bug fix could be back ported to 3.0 in addition.

More appropriate as a feature

**Comment by cntrump** [comment]
Add MacCatalyst support

```bash
export IPHONEOS_DEPLOYMENT_TARGET=13.0
export ARCHS="x86_64;arm64"

./Configure maccatalyst-xcrun
make -j
```

**Pull request closed without merging**